pkg/apis/apps/validation/validation.go

```diff
@@ -93,10 +93,15 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path) fi
 		}
 	case apps.RollingUpdateStatefulSetStrategyType:
 		if spec.UpdateStrategy.RollingUpdate != nil {
-			allErrs = append(allErrs,
-				apivalidation.ValidateNonnegativeField(
-					int64(spec.UpdateStrategy.RollingUpdate.Partition),
-					fldPath.Child("updateStrategy").Child("rollingUpdate").Child("partition"))...)
+			if spec.PodManagementPolicy == apps.OrderedReadyPodManagement &&
+				spec.UpdateStrategy.RollingUpdate != nil &&
+				spec.UpdateStrategy.RollingUpdate.MaxUnavailable != nil &&
+				spec.UpdateStrategy.RollingUpdate.MaxUnavailable.IntValue() > 1 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("updateStrategy").Child("rollingUpdate").Child("maxUnavailable"), spec.UpdateStrategy.RollingUpdate.MaxUnavailable,
+					fmt.Sprintf("with value greater than 1 only allowed for podManagementPolicy '%s'", apps.ParallelPodManagement)))
```

Comment:
Should also prevent using a percentage in this case, no?

Reply:
this was very early API days when we were debating this for whether to implement for OrderedReady or not. Finally we decided to add for both, unit tests cover both scenarios. PTAL